src/macaron/slsa_analyzer/checks/provenance_verified_check.py

```diff
@@ -0,0 +1,78 @@
+# Copyright (c) 2024 - 2024, Oracle and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
+
+"""This module adds a Check that checks whether the provenance is verified."""
+import logging
+
+from sqlalchemy import ForeignKey, String
+from sqlalchemy.orm import Mapped, mapped_column
+
+from macaron.database.table_definitions import CheckFacts
+from macaron.slsa_analyzer.analyze_context import AnalyzeContext
+from macaron.slsa_analyzer.checks.base_check import BaseCheck
+from macaron.slsa_analyzer.checks.check_result import CheckResultData, CheckResultType, Confidence, JustificationType
+from macaron.slsa_analyzer.registry import registry
+from macaron.slsa_analyzer.slsa_req import ReqName
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+class ProvenanceVerifiedFacts(CheckFacts):
+    """The ORM mapping for justifications in the provenance verified check."""
+
+    __tablename__ = "_provenance_verified_check"
+
+    #: The primary key.
+    id: Mapped[int] = mapped_column(ForeignKey("_check_facts.id"), primary_key=True)  # noqa: A003
+
+    #: The verified status of the provenance.
+    verified: Mapped[str] = mapped_column(String, nullable=True, info={"justification": JustificationType.TEXT})
```

Comment:
This field doesn't add any value as it is. I think it would make more sense to report the SLSA level here which can be any of 0, 1, 2, 3, 4. If a provenance doesn't exist, according to SLSA [it is level 0](https://slsa.dev/spec/v1.0/levels#build-l0), if it exists but cannot be verified, according to SLSA [it is level 1](https://slsa.dev/spec/v1.0/levels#build-l1). If a provenance is generated by [SLSA GitHub generator](https://github.com/slsa-framework/slsa-github-generator) and verified, it is level 3 due to isolation enhancements, and otherwise it is level 2. We should be able to assign a level based on the `buildType`.

Please also document this reasoning for the user. I think it's best to create a separate page on the website for this check and explain the rules and logic of it, and what the results mean.